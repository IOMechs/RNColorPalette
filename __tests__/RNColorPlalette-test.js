import React from 'react';
import 'react-native';
import RNColorPalette from '../src/RNColorPalette';
import renderer from 'react-test-renderer';

test('RNColorPalette Component renders correctly', () => {
  const tree = renderer.create(<RNColorPalette />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should change the state of the color picker when we toggle the palette', () => {
  const RNColorPaletteComponent = renderer.create(<RNColorPalette />).getInstance();
    let test = !RNColorPaletteComponent.state.picker;
    RNColorPaletteComponent.togglePalette();
  expect(RNColorPaletteComponent.state.picker).toEqual(test);
});

it('should change the state of the selected color when we select a color', () => {
    const RNColorPaletteComponent = renderer.create(<RNColorPalette />).getInstance();
      let test = !RNColorPaletteComponent.state.openColorPicker;
      RNColorPaletteComponent.toggleColorPicker();
    expect(RNColorPaletteComponent.state.openColorPicker).toEqual(test);
});
 