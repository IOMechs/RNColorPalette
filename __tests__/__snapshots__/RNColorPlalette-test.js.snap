// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    onTouchStart={[Function]}
    style={Object {}}
  />
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 50,
              "overflow": "hidden",
              "position": "absolute",
              "width": 720,
            },
            Object {},
            Object {
              "backgroundColor": "#fff",
            },
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingRight": 8,
            }
          }
          horizontal={true}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#6f7370",
                    "borderRadius": 100,
                    "borderWidth": 1.5,
                    "height": 30,
                    "justifyContent": "center",
                    "lineHeight": 30,
                    "marginLeft": 8,
                    "opacity": 1,
                    "width": 30,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "bottom": 4,
                      "color": "#6f7370",
                      "fontSize": 28,
                      "marginBottom": 3,
                      "position": "relative",
                    }
                  }
                >
                  +
                </Text>
              </View>
              <Modal
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                visible={false}
              >
                <RCTScrollView>
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "space-around",
                          "paddingVertical": 30,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                          }
                        }
                      >
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                              },
                              Object {
                                "height": 710,
                                "width": 710,
                              },
                            ]
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../src/color-wheel-picker/color-wheel.png",
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "alignSelf": "center",
                                },
                                Object {},
                              ]
                            }
                          />
                          <View
                            style={
                              Object {
                                "backgroundColor": "#db3232",
                                "borderColor": "#EEEEEE",
                                "borderRadius": 25,
                                "borderWidth": 3,
                                "elevation": 3,
                                "height": 50,
                                "left": 0,
                                "opacity": 0,
                                "position": "absolute",
                                "shadowColor": "rgb(46, 48, 58)",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.8,
                                "shadowRadius": 2,
                                "top": 0,
                                "width": 50,
                              }
                            }
                          />
                        </View>
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          style={
                            Object {
                              "borderColor": "gray",
                              "borderWidth": 1,
                              "height": 40,
                              "marginVertical": 25,
                              "textAlign": "center",
                              "width": 80,
                            }
                          }
                          underlineColorAndroid="transparent"
                          value="#db3232"
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginRight": 20,
                            }
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityStates={Array []}
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#007AFF",
                                      "fontSize": 18,
                                      "padding": 8,
                                      "textAlign": "center",
                                    },
                                  ]
                                }
                              >
                                Add Color
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          accessibilityRole="button"
                          accessibilityStates={Array []}
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#007AFF",
                                    "fontSize": 18,
                                    "padding": 8,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              Cancel
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </Modal>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
